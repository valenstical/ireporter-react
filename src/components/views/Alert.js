import React from 'react';
import PropTypes from 'prop-types';

/**
 * An alert component to be used to display context messages
 * @param {object} { success, title, message }
 * success - if the context of the message is success or error
 * title - the text to display in the title of the alert box
 * message - the text to display in the body of the alert box
 * @returns {object} The generated JSX object
 */
export default function Alert({ success, title, message }) {
  return (
    <div className={`alert${success ? ' alert-success' : ''}`}>
      <h4>{title}</h4>
      <ol className="unstyled-list">
        {message.map(value => <li key={value}>{value}</li>)}
      </ol>
    </div>


  );
}

Alert.propTypes = {
  success: PropTypes.bool,
  title: PropTypes.string,
  message: PropTypes.array
};

Alert.defaultProps = {
  success: false,
  title: '',
  message: [],
};
