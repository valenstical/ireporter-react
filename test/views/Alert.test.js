import React from 'react';
import { shallow } from 'enzyme';
import Alert from '../../src/components/views/Alert';

describe('<Alert>', () => {
  it('should render without crashing for default context', () => {
    const wrapper = shallow(<Alert success={false} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for successful context', () => {
    const wrapper = shallow(<Alert success />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for empty message', () => {
    const wrapper = shallow(<Alert message={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing for non empty message', () => {
    const wrapper = shallow(<Alert message={['hello', 'hi']} />);
    expect(wrapper).toMatchSnapshot();
  });
});
