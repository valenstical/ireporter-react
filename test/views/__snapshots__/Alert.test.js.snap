// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert> should render without crashing for default context 1`] = `
<div
  className="alert"
>
  <h4 />
  <ol
    className="unstyled-list"
  />
</div>
`;

exports[`<Alert> should render without crashing for empty message 1`] = `
<div
  className="alert"
>
  <h4 />
  <ol
    className="unstyled-list"
  />
</div>
`;

exports[`<Alert> should render without crashing for non empty message 1`] = `
<div
  className="alert"
>
  <h4 />
  <ol
    className="unstyled-list"
  >
    <li
      key="hello"
    >
      hello
    </li>
    <li
      key="hi"
    >
      hi
    </li>
  </ol>
</div>
`;

exports[`<Alert> should render without crashing for successful context 1`] = `
<div
  className="alert alert-success"
>
  <h4 />
  <ol
    className="unstyled-list"
  />
</div>
`;
